### PR TITLE
[SMTChecker] Fix internal error on try/catch

### LINKED ISSUE #9016: [SMTChecker] ICE because variable declaration is unknown
## Description

```
pragma experimental SMTChecker;
contract C {
  function g() public returns (uint, function() external, uint)
  {
    try this.g() returns(uint, function() external, uint b) { (b); }
    catch {}
  }
}
```

throws

https://github.com/ethereum/solidity/blob/7e1f26270b92762d0de3235adad9890fbbb869dd/libsolidity/formal/SMTEncoder.cpp#L1771

## Environment

- Compiler version: latest develop

## Steps to Reproduce

```
$ solc <input.sol>
```

### PR DESCRIPTION
**Pull request by leonardoalt**
Fixes https://github.com/ethereum/solidity/issues/9016

### CODE REVIEW

File: libsolidity/formal/SMTEncoder.cpp [hunk 1]
```diff
@@ -274,6 +274,20 @@ bool SMTEncoder::visit(InlineAssembly const& _inlineAsm)
 	return false;
 }
 
+bool SMTEncoder::visit(TryCatchClause const& _try)
+{
+	if (auto params = _try.parameters())
+		for (auto const& var: params->parameters())
```
Comment (mijovic) [inline libsolidity/formal/SMTEncoder.cpp]: This is where I actually don't like auto as type is not obvious. But as codebase is mixed with both using auto and actual type and there are no rules I guess it is fine to use auto :)

**Review by mijovic — DISMISSED** [review]

File: libsolidity/formal/SMTEncoder.cpp (hunk 1 above)
Comment (leonardoalt) [reply libsolidity/formal/SMTEncoder.cpp]: To me this is exactly where auto is nice, and IMO it makes sense for range loops. The type is not written there, but it's inferrable from the name.

**Comment by leonardoalt** [comment]
Build failed weirdly on Windows, rerunning to see if it was a CI problem.

**Comment by stackenbotten** [comment]
There was an error when running `chk_coding_style` for commit `65db45216a2b54470caa9eae3c9f9d8b9f788769`:
```
Error: Trailing whitespace found:
 test/libsolidity/smtCheckerTests/control_flow/try_catch_2.sol:5: catch (bytes memory x) { 

```
Please check that your changes are working as intended.

**Review by chriseth — APPROVED** [review]

**Review by mijovic — APPROVED** [review]

**Pull request merged**